catroid/src/main/java/org/catrobat/catroid/ui/BaseActivity.java

```diff
@@ -51,9 +55,10 @@ protected void onCreate(Bundle savedInstanceState) {
 		titleActionBar = null;
 		returnToProjectsList = false;
 		returnByPressingBackButton = false;
-		Thread.setDefaultUncaughtExceptionHandler(new BaseExceptionHandler());
-		Utils.checkIfCrashRecoveryAndFinishActivity(this);
+		Thread.setDefaultUncaughtExceptionHandler(new BaseExceptionHandler(this));
+		checkIfCrashRecoveryAndFinishActivity(this);
 		getActionBar().setDisplayHomeAsUpEnabled(true);
+		CrashReporter.sendUnhandledCaughtException();
```

Comment:
I like the idea of decoupling crash reporting and crash recovering a lot. It also makes it easier to test. https://github.com/Catrobat/Catroid/pull/2366#discussion_r125107704

@nishantt95, for now please move this method back to the checkIfCrashRecovery. Moving `CrashReporter.sendUnhandledCaughtException();` somewhere else is out of the scope of this ticket anyway.